src/dataecon/DataEcon.jl

```diff
@@ -189,6 +211,31 @@ function delete_object(de::DEFile, id::C.obj_id_t)
     return nothing
 end
 
+# import ..LittleDict
+
+"""
+    get_all_attributes(de, obejct_id)
+
+Retrieve the names and value of all attributes of the object with the given id.
+They are returned in a dictionary.
+"""
+function get_all_attributes(de::DEFile, id::C.obj_id_t; delim="‖")
```

Comment:
Here is the query that actually runs in `de_get_all_attributes()`: 
![image](https://github.com/bankofcanada/TimeSeriesEcon.jl/assets/12414774/15ae15d8-f86e-4cf6-bd77-ee339e0a75f9)

So, yes, it concatenates all names in one string with the given delimiter and all the values in the same order as the names and with the same delimiter in another string, and it also returns the number of attributes as a separate value.  The main point here is that this query returns at most one row. It may be less efficient than simply ```SELECT `name`, `value` FROM `attributes` WHERE `id` = ?```, but:
* the simple query requires a more complex logic to be coded by the caller -- they'd have to write a loop and additional logic for all corner cases.  Compare with `I._list_catalog()` to see what the user would have to do to iterate the rows of a table returned by SQL. 
* the simple query doesn't distinguish between an object that doesn't exist and an object that exists, but doesn't have any attributes - both cases return a table with 0 rows.  The other query returns 0 rows if object doesn't exist and 1 row (with count=0, names="", values="") if object exists but doesn't have any attributes.

Btw, the case of 0-rows is handled by the library - `de_get_all_attributes` returns an error code (`DE_OBJ_DNE`), which here in the Julia connector is handled by `I._check()` to throw an exception.  So the end user doesn't even have to check for that.